File: WebApplication/src/suppliers/dtos/search/supplier-offering-group/smart-specialization-root-group.dto.ts
import { SmartSpecializationGroupDto } from "./smart-specialization-group.dto";

export class SmartSpecializationRootGroupDto {
    id: number;

    code: string;
    name: string;
    nameAlt: string;
    viewOrder: number;

    smartSpecializations: SmartSpecializationGroupDto[] = [];
}

File: Sc.Models/Dtos/Suppliers/Search/SupplierOfferingGroup/SmartSpecializationRootGroupDto.cs
using Sc.Models.Entities.Base;

namespace Sc.Models.Dtos.Suppliers.Search.SupplierOfferingGroup
{
    public class SmartSpecializationRootGroupDto : Entity
    {
        public string Code { get; set; }
        public string Name { get; set; }
        public string NameAlt { get; set; }
        public int ViewOrder { get; set; }

        public List<SmartSpecializationGroupDto> SmartSpecializations { get; set; } = new List<SmartSpecializationGroupDto>();
    }
}
